pkg/sql/grant_revoke.go

```diff
@@ -209,6 +209,14 @@ func (n *changePrivilegesNode) startExec(params runParams) error {
 			n.changePrivilege(privileges, grantee)
 		}
 
+		// Ensure superusers have exactly the allowed privilege set.
+		// Postgres does not actually enforce this, instead of checking that
+		// superusers have all the privileges, Postgres allows superusers to
+		// bypass privilege checks.
+		if err := privileges.ValidateSuperuserPrivileges(descriptor.GetID(), n.grantOn); err != nil {
```

Comment:
Made a change so ValidateSuperuserPrivileges is a regular check that returns a non assertion error

However Validate calls ValidateSuperuserPrivileges and returns it as an assertion error. Reason being when we were granting/revoking, Validate was the only place we checked superuser privileges and if they're valid, outright replacing that check with assertion errors was incorrect.